src/unstructured_client/utils/_human_split_pdf.py

```diff
@@ -0,0 +1,137 @@
+import copy
+import io
+import logging
+import os
+import functools
+from typing import Optional, Tuple, Callable
+from concurrent.futures import ThreadPoolExecutor
+
+from pypdf import PdfReader, PdfWriter
+
+from unstructured_client import utils
+from unstructured_client.models import shared, operations
+
+logger = logging.getLogger('unstructured-client')
+
+
+def handle_split_pdf_page(func: Callable) -> Callable:
+    """
+    A decorator for splitting PDF by pages and sending each of them separately to backend.
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> operations.PartitionResponse:
+        if len(args) > 0:
+            request = args[1]
+        else:
+            raise ValueError("Expected a request argument for the partition function.")
+        split_pdf_page = request.split_pdf_page
+
+        if not split_pdf_page:
+            return func(*args, **kwargs)
+
+        pages = get_pdf_pages(request.files.content)
+
+        try:
+            call_threads = int(os.getenv("UNSTRUCTURED_CLIENT_SPLIT_CALL_THREADS", "5"))
+        except ValueError:
+            call_threads = 5
+            logger.error("UNSTRUCTURED_CLIENT_SPLIT_CALL_THREADS has invalid value.")
+        logger.info("Splitting PDF by page on client. Using %d threads when calling API.", call_threads)
+        logger.info("Set UNSTRUCTURED_CLIENT_SPLIT_CALL_THREADS env var if you want to change that.")
+
+        results = []
+        with ThreadPoolExecutor(max_workers=call_threads) as executor:
+            if len(args) < 3:
+                retries = None
+            else:
+                retries = args[2]
+            self = args[0]
+
+            call_api_partial = functools.partial(
+                call_api,
+                func=func,
+                self=self,
+                request=request,
+                retries=retries
+            )
+
+            for result in executor.map(call_api_partial, pages):
```

Comment:
Just for future:
If we'll like to report progress in future, it'd be better to use `as_completed` instead of `map` and recreate the order of processing (can use some dict, rather simple).
`map` is not showing the real processing ratio, it's likely to work in "spikes" (because it's preserving the order of input)

Reply:
Ok, good point. I think about creating new issue for adding callback, as for sure won't be able to do that in this one (about a week till we know hook solution for Speakeasy). And your suggestion will be implemented there.